core/src/test/java/org/kohsuke/stapler/DispatcherTest.java

```diff
@@ -273,6 +273,30 @@ public void testRequirePostOnBase() throws Exception {
         assertEquals(1, requirePostOnBase.hit);
     }
 
+    public void testOverloads() throws Exception {
+        TextPage p = new WebClient().getPage(new URL(url, "overloaded/x"));
+        assertEquals("doX(StaplerRequest)", p.getContent().trim());
+    }
+    public final Object overloaded = new Overloaded();
+    public static class Overloaded {
+        @Deprecated
+        public HttpResponse doX() {
+            return HttpResponses.plainText("doX()");
+        }
+        public HttpResponse doX(StaplerRequest req) {
+            return HttpResponses.plainText("doX(StaplerRequest)");
+        }
+        public HttpResponse doX(StaplerResponse rsp) {
+            return HttpResponses.plainText("doX(StaplerResponse)");
+        }
+        public HttpResponse doX(StaplerRequest req, StaplerResponse rsp) {
+            return HttpResponses.plainText("doX(StaplerRequest, StaplerResponse)");
+        }
+        @WebMethod(name = "x")
```

Comment:
Then deprecate your other overloads. Again, the exact ordering of (nondeprecated) methods is arbitrary, so long as it is predictable so any field errors can be reproduced by developers. If you have more than one, you are in the wrong.